packages/grz-common/src/grz_common/transfer.py

```diff
@@ -76,3 +79,26 @@ def init_s3_resource(s3_options: S3Options) -> S3ServiceResource:
     )
 
     return s3_resource
+
+
+def get_version_info(s3_options: S3Options, version_file_path: str) -> str | None:
+    """Download the version file from S3 and return its contents."""
+
+    class VersionFile(BaseModel):
+        minimal_version: version.Version = Field(..., description="Minimum supported version")
+        latest_version: version.Version = Field(..., description="Latest available version")
+
+    try:
+        s3_client = init_s3_client(s3_options)
+
+        # download the version file content directly to memory
+        response = s3_client.get_object(Bucket=s3_options.bucket, Key=version_file_path)
+        version_content = response["Body"].read().decode("utf-8").strip()
+        version_data = json.loads(version_content)
+
+        version_file = VersionFile(**version_data)
+        return version_file
+    
+    except (KeyError, ValidationError, json.JSONDecodeError) as e:
```

Comment:
I think we may want to differentiate between different kinds of errors and discuss what the behaviour should be (i.e., propagate error or make it silent and return None?):
- if the file is not available on the remote: do X
- if the file is broken ( → validation / decode error): do Y
- other: reraise